activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -50,7 +50,23 @@ def process_arguments(owner, args, check_for_additional_values)
               owner.deterministic_encrypted_attributes&.each do |attribute_name|
                 type = owner.type_for_attribute(attribute_name)
                 if !type.previous_types.empty? && value = options[attribute_name.to_s]
-                  options[attribute_name] = process_encrypted_query_argument(value, check_for_additional_values, type)
+                  options[attribute_name.to_s] = process_encrypted_query_argument(value, check_for_additional_values, type)
+                end
+              end
+
+              if owner.respond_to?(:joins_values)
+                owner.joins_values.each do |join_value|
+                  next unless join_value.is_a?(Symbol)
```

Comment:
This currently only supports shallow joins. E.g. `Tag.joins(posts: :author).where(author: { ... })` would not work. Gonna think on how to better support deep joins.